--- mythvideo-scanner/mythvideo-scanner.pl     2011-03-06 10:23:58.000000000 +0100
+++ mythvideo-scanner/mythvideo-scanner.pl     2011-07-28 13:48:48.770182553 +0200
@@ -162,7 +162,7 @@
 
 
 #don't decend beyond these directory names.
-my @blacklist = ('sample','subs','VIDEO_TS','proof');
+my @blacklist = ('sample','subs','VIDEO_TS','proof','BDMV','EXTRAS');
 
 # Scene markers that help find info in the filename
 my $markers;
m
