--- nftytool.c.org	2012-05-23 12:44:50.198179161 +0200
+++ nftytool.c	2012-05-23 12:45:27.077180291 +0200
@@ -114,7 +114,7 @@
 		exit_usage(argv[0], -1);
 	}
 
-	if (config_read_file("nftytool.conf"))
+	if (config_read_file("/etc/nftytool/nftytool.conf"))
 		return -1;
 	
 	config_get_plugins_path(cfile);
--- nftytool.conf.org	2012-05-23 12:50:02.974188743 +0200
+++ nftytool.conf	2012-05-23 12:50:21.661189316 +0200
@@ -1,2 +1,2 @@
 led_idle_brightness 2
-plugins_path plugins
+plugins_path /usr/share/nftytool/plugins
